src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -150,25 +199,45 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
           });
           break;
         case FetchStatus.LOADING:
-          dispatchSidebarStateAction({
-            type: DiscoverSidebarReducerActionType.DOCUMENTS_LOADING,
-            payload: {
-              isPlainRecord: isPlainRecordType,
-            },
-          });
+          console.log('*** fetchStatus.LOADING is firing ', isPlainRecordType);
+          if (isPlainRecordType || !props.useNewFieldsApi) {
+            console.log('*** BOO OLDDDD');
+            dispatchSidebarStateAction({
+              type: DiscoverSidebarReducerActionType.DOCUMENTS_LOADING,
+              payload: {
+                isPlainRecord: isPlainRecordType,
+              },
+            });
+          } else {
+            console.log('*** YES NEEEWWW');
+            fetchFields();
+          }
           break;
         case FetchStatus.COMPLETE:
-          dispatchSidebarStateAction({
-            type: DiscoverSidebarReducerActionType.DOCUMENTS_LOADED,
-            payload: {
-              dataView: selectedDataViewRef.current,
-              fieldCounts: isPlainRecordType
-                ? EMPTY_FIELD_COUNTS
-                : calcFieldCounts(documentState.result),
-              textBasedQueryColumns: documentState.textBasedQueryColumns,
-              isPlainRecord: isPlainRecordType,
-            },
-          });
+          if (!isPlainRecordType || !props.useNewFieldsApi) {
```

Comment:
I should be able to remove these `props.useNewFieldsApi` checks - I think I mixed them up with another config setting.